dns-server/src/storage.rs

```diff
@@ -123,6 +125,7 @@ pub struct Store {
     db: Arc<sled::Db>,
     keep: usize,
     updating: Arc<Mutex<Option<UpdateInfo>>>,
+    poisoned: AtomicBool,
```

Comment:
I think this needs to be `Arc<AtomicBool>`; otherwise different `Store` clones will have distinct instances and won't see each others' stores.